### PR TITLE
Port `EvaluatorTestSuite.kt` tests

### LINKED ISSUE #25: Define default evaluation options for evaluation tests
Need to decide on the default evaluation option behavior for the conformance tests. The most relevant at the moment would be permissive vs strict/type-checking mode.

### PR DESCRIPTION
**Pull request by alancai98**
#8 -- ports tests from `partiql-lang-kotlin`'s [EvaluatorTestSuite.kt](https://github.com/partiql/partiql-lang-kotlin/blob/6f47a3010eab5c7f005bca989664e288a61ab267/lang/test/org/partiql/lang/eval/EvaluatorTestSuite.kt).

Tests ported using the following process:
1. Use [script](https://github.com/alancai98/partiql-lang-kotlin/blob/eval-test-converter/lang/test/org/partiql/lang/eval/EvaluatorTestSuiteToIon.kt) that converts EvaluatorTestSuite.kt tests to the evaluation conformance test format
- Script reads data from the Kotlin test data class and stores as Ion
- Reformats some test fields (e.g. `$partiql_bag` -> `$bag`, remove new lines)
2. Move tests and environments to separate test files
3. Document tests in `partiql-tests` that may need omitted or need to be looked at before adding. E.g. due to not in spec or different compile/evaluation option

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by alancai98** [comment]
Following #25 (defines the evaluation modes), will need to update the ported tests and include the additional `evalMode` in the assertions.

Following #30 (ports the partiql spec tests + restructures `eval` test directories), will also need to update the test directories.

File: partiql-tests-data/eval/misc.ion [hunk 1]
```diff
@@ -0,0 +1,232 @@
+envs::{
```
Comment (alancai98) [inline partiql-tests-data/eval/misc.ion:1]: The formatting of the text `.ion` files in this PR use IonJava's default text pretty printer. This is different than the previous formatting used in the ported spec tests PR #30.

Currently working on an Ion linter for these tests as part of #31, which should address any of the `.ion` text style concerns.

File: partiql-tests-data/eval/query/undefined-variable-behavior.ion [hunk 2]
```diff
@@ -0,0 +1,71 @@
+undefined_variable_behavior::[
+  // ported tests run using UndefinedVariableBehavior.MISSING option.
+  // https://github.com/partiql/partiql-lang-kotlin/blob/main/lang/src/org/partiql/lang/eval/CompileOptions.kt#L27-L36
+  // EvalModeCoerce will return missing for undefined variable references. EvalModeError will give an error for
```
Comment (alancai98) [inline partiql-tests-data/eval/query/undefined-variable-behavior.ion:4]: This set of tests was testing the Kotlin implementation using the `UndefinedVariableBehavior` [option](https://github.com/partiql/partiql-lang-kotlin/blob/main/lang/src/org/partiql/lang/eval/CompileOptions.kt#L27-L36). Undefined variable behavior wasn't explicitly defined in the spec (other than for missing attributes), but I believe `EvalModeCoerce` corresponds to `UndefinedVariableBehavior.MISSING` (undefined variables return `MISSING`) and `EvalModeError` corresponds to `UndefinedVariableBehavior.ERROR` (undefined variables return an `ERROR`).

**Review by am357 — APPROVED** [review]
Very nice work on using a script for porting the tests in; considering using the script I had a cursory review of the tests, overall it LGTM. The fact that at some point in future we'll going to test against these tests from implementations including Kotlin, we should be able to spot any subtle bugs in the tests then.

**Comment by am357** [comment]
> #8 -- ports tests from `partiql-lang-kotlin`'s [EvaluatorTestSuite.kt](https://github.com/partiql/partiql-lang-kotlin/blob/6f47a3010eab5c7f005bca989664e288a61ab267/lang/test/org/partiql/lang/eval/EvaluatorTestSuite.kt).
> 
> Tests ported using the following process:
> 
>     1. Use [script](https://github.com/alancai98/partiql-lang-kotlin/blob/eval-test-converter/lang/test/org/partiql/lang/eval/EvaluatorTestSuiteToIon.kt) that converts EvaluatorTestSuite.kt tests to the evaluation conformance test format
> 
> 
>     * Script reads data from the Kotlin test data class and stores as Ion
> 
>     * Reformats some test fields (e.g. `$partiql_bag` -> `$bag`, remove new lines)
> 
> 
>     2. Move tests and environments to separate test files
> 
>     3. Document tests in `partiql-tests` that may need omitted or need to be looked at before adding. E.g. due to not in spec or different compile/evaluation option
> 
> 
> By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Curious have you then manually split `evaluator_test_suite.ion` output from the script?

**Comment by alancai98** [comment]
> Curious have you then manually split `evaluator_test_suite.ion` output from the script?

Yeah, the script originally ported the `EvaluatorTestSuite.kt` tests to one file. I decided to move the tests to separate files by the namespace/group for better organization. It wasn't too much manual effort other than checking the evaluation environments were in the correct locations.

**Review by jpschorr — APPROVED** [review]
Looks good; nice extraction script.

 I agree with @am357 that running these against the kotlin implementation from which they were extracted is next step in gaining confidence in them; It is infeasible to visually inspect 4kSLOC of test cases and be sure I'm mentally applying the correct operational semantics.

**Pull request merged**